file-formats/tabl/type/zif_aff_indx_v1.intf.abap

```diff
@@ -0,0 +1,71 @@
+INTERFACE zif_aff_indx_v1
+  PUBLIC.
+
+  TYPES:
+    "! <p class="shorttext">Index Fields</p>
+    "! Index fields
+    BEGIN OF ty_indexfield,
+      "! <p class="shorttext">Field Name</p>
+      "! Field name
+      "! $required
+      field_name TYPE c LENGTH 30,
+    END OF ty_indexfield,
+    tt_index_fields TYPE TABLE OF ty_indexfield WITH DEFAULT KEY
+      WITH UNIQUE SORTED KEY name_unique COMPONENTS field_name.
+
+  TYPES:
+    "! <p class="shorttext">ABAP DDIC Secondary Index Properties</p>
+    "! Object type INDX
+    BEGIN OF ty_main,
+      "! $required
+      format_version        TYPE if_aff_types_v1=>ty_format_version,
+      "! <p class="shorttext">Header</p>
+      "! Header
+      "! $required
+      header                TYPE if_aff_types_v1=>ty_header_60,
+      "! <p class="shorttext">Text Index</p>
+      "! Standard- or text-Index
+      "! $required
+      text_index            TYPE abap_bool,
+      "! <p class="shorttext">Unique Index</p>
+      "! Unique flag
+      "! $required
+      unique_index          TYPE abap_bool,
+      "! <p class="shorttext">No DB Index</p>
+      "! Index not on DB, only for ABAP table buffer
+      no_db_index           TYPE abap_bool,
+      "! <p class="shorttext">Fields</p>
+      "! Index fields
+      index_fields          TYPE tt_index_fields,
+      "
+      " the following are attributes for the fulltext index,
+      " available in HANA onPrem only
+      "
+      "! <p class="shorttext">Language Column for Full-Text Index</p>
+      fti_langu_column      TYPE c LENGTH 30,
+      "! <p class="shorttext">Column for Mime Type</p>
+      fti_mime_type_col     TYPE c LENGTH 30,
```

Comment:
Wouldn't it make sense to avoid abbreviations like `col`, `langu` or `indx` later and replace them by `column`, `language` or `index`?

Reply:
In general probably yes, here in this context no. These are the column names that are used to store these FTI properties in a DDIC metadata table. As these names are already complicated, I would prefer to have a 1:1 mapping to the existing column names here and not change them again.

Reply:
Mmmh, but this is actually the intension of the ABAP file formats: Remove the complexity for developers (end users) working with the object type. They shouldn't know the field names on the DB table.

Reply:
Still disagree. Someone needs to map these field names to the appropriate values in the ABAP DDIC. If this is no longer a one-to-one mapping, you add complexity instead of reducing it. I admit that this is a little bit "seen from my point of view"..

Reply:
Maybe, I get you wrong. Wouldn't there be still a 1:1 relation ship if we just rename an field in the file format? E.g., rename field `fti_langu_column` to something like `language_column`.